### ISSUE TITLE
Binary Power in C++

### ISSUE THREAD

**Issue by mohnishdeshpande** [MakeContributions/DSA#808]
## Information about Algorithm

Faster computing of powers (a^b) using Binary Exponentiation.
### Have you read the [Contributing.md](https://github.com/MakeContributions/DSA/blob/main/CONTRIBUTING.md) and [Code of conduct](https://github.com/MakeContributions/DSA/blob/main/CODE_OF_CONDUCT.md)

- [x] Yes
- [ ] No

**Comment by SubhamPanda2003** [comment]
Hey,I am willing to work on this, please assign it to me.

**Comment by ming-tsai** [comment]
Hi @mohnishdeshpande, @angshudas and @Arsenic-ATG,
We have both pull requests #810 and #815 working on this, please make sure which should be closed and make pair programming in another pull request.

**Comment by mohnishdeshpande** [comment]
Hi @ming-tsai,
In #815, there are a fixes changes required.
On line 16:
-long ans= power(base , expo/2);
+long ans= power_recursion(base , expo/2);
and line 33:
-res = res * expo;
+res = res * base;

To solve the issue here, I have added the recursive implementation from @angshudas to #810 , sourcing his account in comments.

You can review and merge #810 if you are happy with it.
P.S. I have tested it already.

**Comment by Arsenic-ATG** [comment]
> Hi @mohnishdeshpande, @angshudas and @Arsenic-ATG,
> We have both pull requests #810 and #815 working on this, please make sure which should be closed and make pair programming in another pull request.

According to me, the one opened earlier should be given preference (i.e.  #810 ). 

@angshudas , if you want to make changes, then a preferable way might be to work on source branch of previous pull request ( [mohnishdeshpande:808](https://github.com/mohnishdeshpande/DSA/tree/808) in this case ). It will keep the changes regarding one algorithm isolated to one PR only (at least on the main repository).

Or 

Update the existing file once the previous one is merged (as refactoring/optimisation)

**Comment by Arsenic-ATG** [comment]
> To solve the issue here, I have added the recursive implementation from @angshudas to #810 , sourcing his account in comments.

I guess a much better alternative would be to mark @angshudas as a co-author of the commit adding recursive version of the algorithm, like this, version control will also be able to keep track of original author of the changes in the repo ( would be helpful if someone does a `git blame` on the file )
